lib/src/test/java/uk/gov/di/ipv/core/library/service/UserIdentityServiceTest.java

```diff
@@ -120,19 +122,57 @@ void shouldSetVotClaimToP2OnSuccessfulIdentityCheck()
 
         UserIdentity credentials =
                 userIdentityService.generateUserIdentity(
-                        "user-id-1", "test-sub", "P2", currentVcStatuses);
+                        USER_ID_1, "test-sub", "P2", currentVcStatuses);
 
         assertEquals(VectorOfTrust.P2.toString(), credentials.getVot());
     }
 
+    @Test
+    void checkValidateBirthDatesInCredentialsSuccess() throws HttpResponseExceptionWithErrorBody {
+        List<VcStoreItem> vcStoreItems =
+                List.of(
+                        createVcStoreItem(USER_ID_1, "ukPassport", SIGNED_VC_2, Instant.now()),
+                        createVcStoreItem(USER_ID_1, "dcmaw", SIGNED_VC_3, Instant.now()));
+
+        List<VcStatusDto> currentVcStatuses = List.of(new VcStatusDto("test-issuer", true));
+
+        when(userIdentityService.getVcStoreItems(USER_ID_1)).thenReturn(vcStoreItems);
+        when(mockConfigService.getComponentId(any())).thenReturn("test-issuer");
+
+        boolean isValid =
+                userIdentityService.checkBirthDateCorrelationInCredentials(
+                        USER_ID_1, currentVcStatuses);
+
+        assertTrue(isValid);
+    }
+
+    @Test
+    void checkValidateBirthDatesInCredentialsFalse() throws HttpResponseExceptionWithErrorBody {
```

Comment:
This could be more descriptive eg. "checkBirthDateCorrelationInCredentialsReturnsFalseWhenBirthDatesDiffer" (and above)